pageserver/src/basebackup.rs

```diff
@@ -80,7 +80,12 @@ where
         let (backup_prev, backup_lsn) = if let Some(req_lsn) = req_lsn {
             // Backup was requested at a particular LSN. Wait for it to arrive.
             info!("waiting for {}", req_lsn);
-            timeline.wait_lsn(req_lsn)?;
+
+            let ctx = format!(
+                "Backup was already requested at lsn {}. Waiting for it to arrive.",
+                req_lsn
+            );
+            timeline.wait_lsn(req_lsn, &ctx)?;
```

Comment:
We already have basebackup context, it's captured by our logger. Take a look at this log for example:

```
2022-08-15T09:32:03.915134Z ERROR query handler for 'basebackup 8c39fbeaa128d63bc0b2ce5918e55f78 ae25ca1d50b9d17d112d39f1187d991b 0/1845070' failed: Timed out while waiting for WAL record at LSN 0/1845070 to arrive, last_record_lsn 0/17FA000 disk consistent LSN=0/16B5A38
Caused by:
    SeqWaitError
Stack backtrace:
   0: <pageserver::layered_repository::timeline::LayeredTimeline as pageserver::repository::Timeline>::wait_lsn::{{closure}}
             at /home/runner/pageserver/src/layered_repository/timeline.rs:430:16
      prometheus::histogram::Histogram::observe_closure_duration
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/prometheus-0.13.1/src/histogram.rs:716:19
      <pageserver::layered_repository::timeline::LayeredTimeline as pageserver::repository::Timeline>::wait_lsn
             at /home/runner/pageserver/src/layered_repository/timeline.rs:429:9
   1: pageserver::basebackup::Basebackup<W,T>::new
             at /home/runner/pageserver/src/basebackup.rs:83:13
...
```

The same also applies to other `wait_lsn` callers, because we have a full stacktrace in the logs.

Also note that passing context as a string can be replaced with calling [with_context](https://docs.rs/anyhow/1.0.36/anyhow/trait.Context.html#tymethod.with_context), which can look like:

```rust
timeline.wait_lsn(req_lsn).with_context(|| "waiting for basebackup")?;
```

Reply:
Hi, @petuhovskiy thanks for your comments.
While implementation itself I was unsure, but it seemed way too approachable so I gave it a try.

I will close this one and check the linked code. Thank you.